pkg/utils/grpc.go

```diff
@@ -0,0 +1,32 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+)
+
+func ParseEndpoint(ep string) (string, string, error) {
+	if strings.HasPrefix(strings.ToLower(ep), "unix://") || strings.HasPrefix(strings.ToLower(ep), "tcp://") {
+		s := strings.SplitN(ep, "://", 2)
+		if s[1] != "" {
+			return s[0], s[1], nil
+		}
+	}
+	return "", "", fmt.Errorf("invalid endpoint: %v", ep)
+}
+
+func LogGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	klog.V(2).Infof("GRPC call: %s", info.FullMethod)
+	klog.V(2).Infof("GRPC request: %v", req)
```

Comment:
we should increase the verbosity of these logs as it can contain sensitive data. we should also see if we need to mask anything here.

Reply:
The request and response for this doesn't contain much information other than the plain and cipher text. Both are information that we don't want to log at any levels. So I've removed that and added logs in the methods to log start and end.